gossip/src/crds_value.rs

```diff
@@ -204,6 +209,47 @@ impl SnapshotHash {
         }
     }
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, AbiExample)]
+pub struct IncrementalSnapshotHash {
+    pub from: Pubkey,
+    pub base_slot: Slot,
+    pub hashes: Vec<(Slot, Hash)>,
+    pub wallclock: u64,
+}
+
+impl Sanitize for IncrementalSnapshotHash {
+    fn sanitize(&self) -> Result<(), SanitizeError> {
+        sanitize_wallclock(self.wallclock)?;
+        if self.base_slot >= MAX_SLOT {
+            return Err(SanitizeError::ValueOutOfBounds);
+        }
+        if self.hashes.len() > MAX_SNAPSHOT_HASHES {
+            return Err(SanitizeError::ValueOutOfBounds);
+        }
+        for (slot, _) in &self.hashes {
+            if *slot >= MAX_SLOT {
+                return Err(SanitizeError::ValueOutOfBounds);
+            }
+            if self.base_slot >= *slot {
+                return Err(SanitizeError::InvalidValue);
+            }
+        }
+        self.from.sanitize()
+    }
+}
+
+impl IncrementalSnapshotHash {
+    pub fn new(from: Pubkey, base_slot: Slot, hashes: Vec<(Slot, Hash)>) -> Self {
+        Self {
+            from,
+            base_slot,
+            hashes,
+            wallclock: timestamp(),
```

Comment:
I would rather take `wallclock` as an argument rather than pulling it here. e.g. it would be more suitable for testing.

Reply:
Done! I've removed the `new()` fn, since every field is passed directly to the struct instance.